model/src/main/scala/ch.epfl.scala.index.model/release/BackendVersion.scala

```diff
@@ -0,0 +1,98 @@
+package ch.epfl.scala.index.model
+package release
+
+import fastparse.NoWhitespace._
+import fastparse._
+
+sealed trait BackendVersion {
```

Comment:
It would be nice to have some scaladocs explaining what this sealed trait models, and what `family` and `isValid` are.

Reply:
Also, I’m not sure `BackendVersion` is a good name for that. I think “binary version” or “scala version” would be better.

Reply:
Agreed for `ScalaVersion` and for the scaladocs

Reply:
Ok for the scaladoc

Reply:
I dont think that `BinaryVersion` is a better name because there already is a `BinaryVersion` ADT that is used for describing the scala-js version and the scala-native version

Reply:
And the problem with `ScalaVersion` is to find a name of the two subtypes that are currently named `ScalaBackend` and `DottyBackend`. Is 0.23 a ScalaVersion? My answer would be not yet because dotty is not Scala 3 yet. But still dotty can be used as a backend for the javascript and scala-native compilers. That is why I choosed the name `BackendVersion`. A scala js target is made of a `BackendVersion` and a javascript `BinaryVersion`. It makes sense to me.

Reply:
*Backend* is definitely the wrong word. If you want to reason along your reasoning, it should be *frontend* version. Scala 2/Dotty are frontends, not backends. Backends would be JVM/JS/Native. But even there for JS itself we have a front-end and a back-end. It's all very relative. But in any case the distinction scalac/dotty is the very first "end" of the language, so in any naming scheme it would be a frontend, not a backend.

But I think the best name for this---independently of the philosophical debate whether Dotty is "Scala" or not---is `LanguageVersion`. This definitely represents a version of the *language*, whether that language is called "Scala" or "Dotty". So I would recommend `LanguageVersion` with the subtypes `ScalaVersion` and `DottyVersion`.

Reply:
You are totally right. I have inverted the meaning of Backend and Frontend in my head. `LangugageVersion` as you suggest is a better name.